src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Metadata/DefaultModelMetadataProvider.cs

```diff
@@ -84,6 +93,13 @@ private ModelMetadataCacheEntry CreateCacheEntry(ModelMetadataIdentity key)
             return new ModelMetadataCacheEntry(metadata, details);
         }
 
+        private ModelMetadata GetMetadataForObjectType()
+        {
+            var key = ModelMetadataIdentity.ForType(typeof(object));
+            var entry = CreateCacheEntry(key);
```

Comment:
> Wondering if we can reuse the Object Metadata if any of the properties are of type `object`?

No, property metadata is a completely different beast from type metadata. I'm wondering if we can save and reuse this `ModelMetadataCacheEntry` for when anything calls `GetMetadataForProperties(typeof(object))`. See [line 49](https://github.com/aspnet/Mvc/pull/4518/files#diff-c8fa65ecc8077ca10cd5f923c0082de2R49).